x-pack/plugins/security_solution/scripts/run_cypress/parallel_serverless.ts

```diff
@@ -527,6 +532,16 @@ ${JSON.stringify(cypressConfigFile, null, 2)}
               }
               process.env.TEST_CLOUD_HOST_NAME = new URL(BASE_ENV_URL).hostname;
 
+              if (argv.beforeSpec) {
+                const beforeSpecFilePath = require.resolve(`../../${argv.beforeSpec}`) as string;
+                const {
+                  beforeSpec,
+                }: {
+                  beforeSpec: (config: typeof cyCustomEnv) => Promise<void>;
+                } = await import(beforeSpecFilePath);
```

Comment:
Runtime checks could be a better approach here instead of type casting. node.js's `await import()` returns `any` which can be casted to anything. In fact `beforeSpecFilePath` might return a module without exported `beforeSpec` member or it could be not a function. The code could look like

```ts
const module: unknown = await import(beforeSpecFilePath);

if (typeof module !== 'object' || module === null) {
  throw new Error(
    `${chalk.bold(
      beforeSpecFilePath
    )} expected to explicitly export function member named "beforeSpec"`
  );
}

if (!('beforeSpec' in module) || typeof module.beforeSpec !== 'function') {
  throw new Error(
    `${chalk.bold('beforeSpec')} exported from ${chalk.bold(
      beforeSpecFilePath
    )} is not a function`
  );
}

await module.beforeSpec();
```

Reply:
Makes total sense, thank you! 
https://github.com/elastic/kibana/pull/189860/commits/09eabd0f6de71b48837f1065164f04a9ca6c7fff